site/app/views/admin/AdminGradeableView.php

```diff
@@ -84,7 +122,16 @@ public function show_add_gradeable($type_of_action, AdminGradeable $admin_gradea
             "course"          => $_GET['course'],
 
             // Graders Page Specific
-            "all_graders"    => $graders
+            "all_graders"    => $graders,
+            //Repository name
+            "repo_name"        => $repo_name,
+            //All the uploaded config paths
+            "all_repository_paths"    => $all_repository_paths,
+            "all_uploaded_paths"      => $all_uploaded_paths,
+            "default_paths"           => $default_paths,
+            "which_option"            => $which_option,
+            //build outputs
+            "cmake_output"            => htmlentities($cmake_output)
```

Comment:
Either package all of these options (line 126 to 134) into a class or given them better names